scmrepo/fs.py

```diff
@@ -54,22 +52,14 @@ def __init__(
             resolved = resolver(scm, rev or "HEAD")
             tree_obj = scm.pygit2.get_tree_obj(rev=resolved)
             trie = GitTrie(tree_obj, resolved)
-            path = scm.root_dir
-        else:
-            assert path
 
         self.trie = trie
-        self.root_dir = path
         self.rev = self.trie.rev
 
     def _get_key(self, path: str) -> Tuple[str, ...]:
-        from scmrepo.utils import relpath
-
-        if os.path.isabs(path):
-            path = relpath(path, self.root_dir)
-        relparts = path.split(os.sep)
-        if relparts == ["."]:
-            return ()
+        relparts = path.split(self.sep)
+        if relparts and relparts[0] in (".", ""):
```

Comment:
Much better to normalize the path beforehand, but our `fs.path` is not yet available in fsspec, so we'll get to it later.